src/wallet/db.cpp

```diff
@@ -826,6 +826,8 @@ void BerkeleyDatabase::Flush(bool shutdown)
             LOCK(cs_db);
```

Comment:
Note that if two or more BerkeleyDatabase share the same BerkeleyEnvironment, the `env` would be a wild pointer after the first call.

Reply:
Good catch. This appears to be a bug introduced in 0b82bac76d0f842bd2294a290388536951fbc576 from #13111 and then tweaked in a769461d5e37ddcb771ae836254fdc69177a28c4 in #12493. It seems not directly related to this issue, so probably best addressed separately.

Reply:
Thread https://github.com/bitcoin/bitcoin/pull/14320#discussion_r220332273

> Good catch indeed. The bug was not introduced in 0b82bac but in a769461?

0b82bac76d0f842bd2294a290388536951fbc576 added the call to erase the environment too early and a769461d5e37ddcb771ae836254fdc69177a28c4 just moved the call, IIUC.